examples/experimental/openvino/ssd_mobilenet_v1_fpn/main.py

```diff
@@ -0,0 +1,112 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Iterable
+from typing import Any
+from pathlib import Path
+from functools import partial
+
+import nncf
+import numpy as np
+import openvino.runtime as ov
+
+from openvino.tools.pot.api.samples.object_detection.data_loader import COCOLoader
+from openvino.tools.pot.api.samples.object_detection.metric import MAP
+
+
+FILE = Path(__file__).resolve()
+# Relative path to the `ssd_mobilenet_v1_fpn` directory.
+ROOT = FILE.parent.relative_to(Path.cwd())
+# Path to the directory where the original and quantized IR will be saved.
+MODEL_DIR = ROOT / 'ssd_mobilenet_v1_fpn_quantization'
+# Path to COCO validation dataset.
+DATASET_DIR = ROOT / 'coco2017' / 'val2017'
+ANNOTATION_FILE = ROOT / 'coco2017' / 'instances_val2017.json'
+
+
+ie = ov.Core()
+
+
+def run_example():
+    """
+    Runs the SSD MobileNetV1 FPN quantize with accuracy control example.
+    """
+    # Step 1: Load the OpenVINO model.
+    ir_model_xml = ROOT / 'public' / 'ssd_mobilenet_v1_fpn_coco' / 'FP32' / 'ssd_mobilenet_v1_fpn_coco.xml'
+    ir_model_bin = ir_model_xml.with_suffix('.bin')
+    ov_model = ie.read_model(model=ir_model_xml, weights=ir_model_bin)
+
+    # Step 2: Create data source.
+    dataset_config = {
+        'images_path': f'{DATASET_DIR}/',
+        'annotation_path': ANNOTATION_FILE,
+    }
+    data_source = COCOLoader(dataset_config)
+
+    # Step 4: Apply quantization algorithm.
+
+    # Define the transformation method. This method should take a data item returned
+    # per iteration through the `data_source` object and transform it into the model's
+    # expected input that can be used for the model inference.
+    def transform_fn(data):
+        images, _ = data
+        return np.expand_dims(images, 0)
+
+    # Wrap framework-specific data source into the `nncf.Dataset` object.
+    calibration_dataset = nncf.Dataset(data_source, transform_fn)
```

Comment:
Do we really need two objects here? Can we just create only one object here, let's say `validation_dataset` and pass it as calibration and validation dataset?

Reply:
Done